File: packages/go/contracts/domain/instrument.go
package domain

type InstrumentID string

type Venue string

type AssetClass string

type Timeframe string

const (
	AssetClassEquity AssetClass = "equity"
	AssetClassOption AssetClass = "option"
	AssetClassFuture AssetClass = "future"
	AssetClassForex  AssetClass = "forex"
	AssetClassCrypto AssetClass = "crypto"
)

type Instrument struct {
	ID            InstrumentID `json:"id"`
	Symbol        string       `json:"symbol"`
	Venue         Venue        `json:"venue,omitempty"`
	AssetClass    AssetClass   `json:"asset_class"`
	BaseCurrency  Currency     `json:"base_currency,omitempty"`
	QuoteCurrency Currency     `json:"quote_currency,omitempty"`
	TickSize      Decimal      `json:"tick_size,omitempty"`
	LotSize       Decimal      `json:"lot_size,omitempty"`
}

File: packages/python/src/trend_buddy_shared/contracts/domain/instrument.py
from dataclasses import dataclass
from decimal import Decimal
from enum import StrEnum


class AssetClass(StrEnum):
    EQUITY = "equity"
    OPTION = "option"
    FUTURE = "future"
    FOREX = "forex"
    CRYPTO = "crypto"


Timeframe = str


@dataclass
class Instrument:
    id: str
    symbol: str
    asset_class: AssetClass
    venue: str = ""
    base_currency: str = ""
    quote_currency: str = ""
    tick_size: Decimal = Decimal("0")
    lot_size: Decimal = Decimal("0")
